### PR TITLE
dev.lapce.lapce: Initial Flathub release

### PR DESCRIPTION
**Pull request by sebastian-de**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  `--filesystem=home` is used to access source files.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
  I contacted Dongdong Zhou via Matrix (#lapce-editor-discord-general:matrix.org)
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
  I didn't submit `dev.lapce.lapce.metainfo.xml` since Lapce is a cross platform application.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by sebastian-de** [comment]
bot, build dev.lapce.lapce

**Comment by bilelmoussaoui** [comment]
>Any additional patches or files have been submitted to the upstream projects concerned. (If not, explain why.)
I didn't submit dev.lapce.lapce.metainfo.xml since Lapce is a cross platform application.

I don't think this matters, Firefox is a crossplatform app yet it has it own metainfo file upstream.

File: dev.lapce.lapce.desktop [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[Desktop Entry]
```
Comment (bilelmoussaoui) [inline dev.lapce.lapce.desktop:1]: Should be contributed upstream

File: dev.lapce.lapce.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
```
Comment (bilelmoussaoui) [inline dev.lapce.lapce.metainfo.xml:2]: Same as the desktop file

Comment (sebastian-de) [reply dev.lapce.lapce.metainfo.xml:2]: Thanks for your fast review! I'll create a PR for these files upstream.

Comment (sebastian-de) [reply dev.lapce.lapce.metainfo.xml:2]: Are `<release>` tags mandatory for a release on flathub? `flatpak run org.freedesktop.appstream-glib validate dev.lapce.lapce.metainfo.xml` does complain if they are not present.
The upstream developers doesn't even use Linux, so I don't want to bother them with updating metainfo.xml with every release.

Comment (hfiguiere) [reply dev.lapce.lapce.metainfo.xml:2]: yes it's required.

File: dev.lapce.lapce.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-stable
+command: lapce
+finish-args:
+  - --device=dri
+  - --filesystem=home
+  - --share=ipc
+  - --share=network
+  - --socket=wayland
+  # Full X11-Support is needed, fallback-x11 doesn't seem to work, yet.
```
Comment (hfiguiere) [inline dev.lapce.lapce.yaml:14]: then remove wayland.

Comment (sebastian-de) [reply dev.lapce.lapce.yaml:14]: The weird thing is that Lapce runs native on Wayland with this configuration. If I remove `--socket=x11` or use `--socket=fallback-x11` Lapce crashes on startup.
This is definitely an upstream bug, but I'd like to keep this configuration until this is fixed.

Comment (hfiguiere) [reply dev.lapce.lapce.yaml:14]: ok.

File: dev.lapce.lapce.metainfo.xml (hunk 2 above)
Comment (sebastian-de) [reply dev.lapce.lapce.metainfo.xml:2]: The files have been merged into the upstream master branch an can be deleted after the next release.

**Comment by sebastian-de** [comment]
bot, build dev.lapce.lapce

**Comment by hfiguiere** [comment]
the prefered license for metadata is `CC0-1.0`. `Apache-2.0` is no valid.

**Comment by sebastian-de** [comment]
The queued build is already with updated metadata license tag: https://github.com/lapce/lapce/pull/515

File: dev.lapce.lapce.metainfo.xml (hunk 2 above)
Comment (sebastian-de) [reply dev.lapce.lapce.metainfo.xml]: done

**Comment by VarLad** [comment]
@barthalion @sebastian-de Is it possible to enable the D-Bus Session socket by default. Its important to be able to use `flatpak-spawn` and use our system applications.

**Comment by hfiguiere** [comment]
please file issues in the right place https://github.com/flathub/dev.lapce.lapce/issues

**Pull request closed without merging**